src/SchematicHQ.Client/Schematic.cs

```diff
@@ -0,0 +1,173 @@
+using OneOf;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace SchematicHQ.Client;
+
+public partial class Schematic
+{
+    private readonly ClientOptions _options;
+    private readonly IEventBuffer<CreateEventRequestBody> _eventBuffer;
+    private readonly ISchematicLogger _logger;
+    private readonly List<ICacheProvider<bool?>> _flagCheckCacheProviders;
+    private readonly bool _offline;
+    public readonly SchematicApi API;
+
+    public Schematic(string apiKey, ClientOptions? options = null)
+    {
+        _options = options ?? new ClientOptions();
+        _offline = _options.Offline;
+        _logger = _options.Logger ?? new ConsoleLogger();
+
+        var httpClient = _offline ? new OfflineHttpClient() : _options.HttpClient;
+        API = new SchematicApi(apiKey, _options.WithHttpClient(httpClient));
+
+        _eventBuffer = _options.EventBuffer ?? new EventBuffer<CreateEventRequestBody>(
+            async items => 
+            {
+                var request = new CreateEventBatchRequestBody
+                {
+                    Events = items
+                };
+                await API.Events.CreateEventBatchAsync(request);
+            },
+            _logger,
+            flushPeriod: options.DefaultEventBufferPeriod  // default flush period
+        );
+        _eventBuffer.Start();
+
+        _flagCheckCacheProviders = _options.CacheProviders ?? new List<ICacheProvider<bool?>>
+        {
+            new LocalCache<bool?>()
+        };
+    }
+
+    public void Shutdown()
+    {
+        _eventBuffer.Dispose();
+    }
+
+    public async Task<bool> CheckFlag(string flagKey, Dictionary<string, string>? company = null, Dictionary<string, string>? user = null)
+    {
+       
+        if (_offline)
+            return GetFlagDefault(flagKey);
+
+        try
+        {
+            string cacheKey = flagKey;
+             if (company != null && company.Count > 0)
```

Comment:
Put something to indicate company- "C" and user "U"

Still include this in teh case where company = null or user = null